File: PSet2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[])
{
    bool input = false;
    int k;
    
    do
    {
        if(argc != 2)
        {
            printf("Usage: ./caesar k \n");
            return 1;
        }
        else
        {
            k = atoi(argv[1]); // convert string argv[] to an integer k (key)
            input = true;
        }
    } while(!input);
    
    string pt;
    
    // prompt user for input 
    do 
    {
        printf("plaintext: ");
        pt = get_string();
    } while (strlen(pt) == 0); // keep asking until there is an input
    
    printf("ciphertext: ");
    
    // iterate through the string
    for (int i = 0, n = strlen(pt); i < n; i++)
    {
        if (isalpha(pt[i]))
        {
            if (islower(pt[i]))
            {
                int ascii = pt[i];
                // to wrap around from 'z' to 'a'
                ascii = ascii - 97;
                ascii = ascii + k;
                ascii = ascii % 26;
                ascii = ascii + 97;
                printf("%c", ascii);
            }
            else 
            {
                int ascii = pt[i];
                // to wrap around from 'Z' to 'A'
                ascii = ascii - 65;
                ascii = ascii + k;
                ascii = ascii % 26;
                ascii = ascii + 65;
                printf("%c", ascii);
            }
        }
        else 
        {
            printf("%c", pt[i]);
        }
    }
        
    printf("\n");
}

File: PSet6/caesar.py
import cs50
import math
import sys

if len(sys.argv) < 2:
    print('Usage: python caesar.py k')
    sys.exit()
    
while True:
    k = sys.argv[1]
    k = int(k)
    
    print('plaintext: ', end="")
    pt = input()
    
    while (len(pt) == 0):
        print('plaintext: ', end="")
        pt = input()
        
    print('ciphertext: ', end="")
    
    for char in pt:
        if(char.isalpha()):
            if(char.islower()):
                ascii = ord(char)
                ascii = ascii - 97
                ascii = ascii + k
                ascii = ascii % 26
                ascii = ascii + 97
                print(chr(ascii), end="")
                
            else:
                ascii = ord(char)
                ascii = ascii - 65
                ascii = ascii + k
                ascii = ascii % 26
                ascii = ascii + 65
                print(chr(ascii), end="")
                
        else:
            print(char, end="")
                
    print('\n', end="")
    break